Add Strimzi incubation proposal (#1)
ppatierno/toc: proposals/incubation/strimzi.md

But running it on Kubernetes can be complex and tedious.
Strimzi makes it easy by using the operator pattern and declarative deployments.
It provides operators not just for deploying Kafka clusters, but also for managing topics, users and connectors to make sure running Apache Kafka feels Kube-native.

Strimzi also integrates with many other CNCF projects such as:

* Kubernetes
* [Prometheus][prom] for metrics and monitoring
* [Open Policy Agent][opa] for authorization
* [KEDA][keda] for autoscaling of Kafka clients
* [Jaeger][jaeger] and [OpenTelemetry][otel] for tracing

In addition, Strimzi is used by [Knative][knative] to provide its Kafka source.

Strimzi can be installed using [Helm][helm] charts or using [OperatorHub.io][operatorhub] which is part of the [Operator Framework][operatorframework] project.

Strimzi joined CNCF Sandbox in September 2019

## Statement on alignment with the CNCF mission

> The Foundation's mission is to make cloud native computing ubiquitous

While Kafka is _capable_ of horizontal scalability, achieving such things as zero-downtime upgrades, and scaling clusters within Kubernetes without an operator is too high a barrier for many organizations.

> This prevents ubiquitous use of Kafka anywhere within the cloud.

Reviewer: _within the cloud_ sounds a bit weird to me. What about something like _with cloud native platforms such as Kubernetes_

Strimzi provides declarative APIs for Apache Kafka resources, including broker clusters, Kafka Connect clusters, connectors, and Kafka MirrorMaker instances (for inter-cluster replication).

> By automating the operational burdens of running Kafka, Strimzi brings this important technology into the hands of anyone using Kubernetes.

Reviewer: This sounds like there is some statement missing about how we help to automate operational burdens.

Author: Can you elaborate a little bit more please? Any suggestions?

Reviewer: It seems disconnected. You talk about declarative APIs and then you talk about how automation helps. But you do not explain how / what Strimzi automates.

Author: Something like this?

> Thanks to the above, Strimzi simplifies the way to deploy a Kafka cluster on Kubernetes as well as helps to automate the day-2 operations like upgrading brokers configuration, Kafka version and security related issues.

@tombentley any better from your side?

Reviewer: That sounds good to me -> it adds the link which is missing currently.

## Community

> In the Strimzi community users and developers can interact in several places: 

Reviewer: Should we add some more stats here? Number of people in the Slack channel, number of contributors etc.?

Author: The number of contributors is reported in the `Commits and merged contributions` section but we can report the people on Slack channel for sure here.

* A public channel on the CNCF slack and mailing list
* Github issues and discussions

> * A public "community call" every other week

Reviewer: Maybe you can add link to the recordings?

* We have participated in numerous KubeCon "Office Hours" events.

Strimzi encourages contributions from the community, and has a good record of merging such contributions.

## Incubation Stage Requirements

<!-- NOTE: 
See [Incubation Stage](https://github.com/cncf/toc/blob/main/process/graduation_criteria.adoc#incubating-stage) requirements. -->

### Production usage

> Strimzi is used in production by a number of organizations (see [ADOPTERS.md](https://github.com/strimzi/strimzi-kafka-operator/blob/main/ADOPTERS.md) for some public references). 

Reviewer: Maybe you can also point to the website?

In addition, the maintainers are aware of further non-public users of both Strimzi and commercially supported distributions of Strimzi. 

### Committers

> Strimzi currently has 7 maintainers, who have commit rights.
> The [governance](https://github.com/strimzi/governance) model also defines area-specific owners, whose reviews count towards PR approval.
> 
> Those maintainers are currently employed by three separate companies: Red Hat, IBM and CN Group.

Reviewer: Maybe we can re-use some stuff from the annual review and make this sound a bit better ...

```suggestion
Strimzi has currently 7 maintainers from three companies (Red Hat, IBM and CN Group).
The full maintainer list can be found in our [GitHub repository (https://github.com/strimzi/governance/blob/master/MAINTAINERS).

Additionally to the maintainers, 8 other contributors have _Write_ permissions to individual repositories and act as owners for the sub-projects within these repositories. 
The full list can be found in our [GitHub repository](https://github.com/strimzi/governance/blob/main/COMPONENT-OWNERS) as well.
```

### Commits and merged contributions

> At the time of writing:

Reviewer: Might be worth adding a month / year here when you took the numbers? It will notbe obvious what the time of writing was.

Author: We can do this when opening the official PR. Maybe we can review the numbers at that time and write `At the time of opening this proposal:` or something like that, instead of a specific date. But I have no strong opinion on which is the best.

Reviewer 2: Suggestion:
_The following figures represent a snapshot of commits and merges to the Strimzi project as of `date-here`._

* 2316 commits in the last 12 months, and more in the last 3 months than the previous 3 months
* 438 contributors in the last 12 months
* more than 3k stars and ~830 forks on github.

See our [devstats](https://all.devstats.cncf.io/d/54/project-health-table?orgId=1&var-repogroup_name=Strimzi) for more details.

### Versioning scheme

> The project's release version in described in [RELEASES.md](https://github.com/strimzi/governance/blob/main/RELEASES.md).

Reviewer: Would it be worth to add two or there sentences here instead of just the raw link? Saying how often we release and that we follow emantic versioning etc.?

Author: I just took a couple of sentences from the linked file.

### Security processes

> The project's security processes are described in [SECURITY.md](https://github.com/strimzi/.github/blob/main/SECURITY.md).

Reviewer: Again, maybe you can add some more stuff here ... at least some general things how committed to security we are and how more details can  be found on the link?

Author: I just took a couple of sentences from the linked file.

## Future Plans

> ### Functionality

Reviewer: Maybe you can also mention stretch clusters?

Author: @tombentley I remember we had that in the first draft but it's gone. Why?

> There are a number of technical features which we want to explore, that will keep Strimzi relevant and increase its applicability.

Reviewer: _explore_ sounds weird to me. I think we want to do all of this, and _explore_ sounds very uncertain to me.

Author: Any better suggestions?

Reviewer: I think you should talk about features we _plan to implement_  or _want to add_. I would also leave out _keep Strimzi relevant_

```suggestion
There are several major features which we work on to further increase Strimzi capabilities.
```

# Strimzi Incubation Stage Proposal

## Introduction

Strimzi is a project focused on running [Apache Kafka][ak] on [Kubernetes][kube].
Apache Kafka is the _de facto_ standard platform for building real-time data pipelines.
It provides support for high-throughput / low-latency messaging, as well as sophisticated development options that cover all the stages of a distributed data streaming pipeline, from ingestion, through storage to processing.
But running it on Kubernetes can be complex and tedious.
Strimzi makes it easy by using the operator pattern and declarative deployments.
It provides operators not just for deploying Kafka clusters, but also for managing topics, users and connectors to make sure running Apache Kafka feels Kube-native.

Strimzi also integrates with many other CNCF projects such as:

> * Kubernetes

Reviewer 3: nit: Kubernetes[kube]

Author: We didn't add the link here because we already did on line 5 at the beginning of the introduction.

Reported vulnerabilities will be investigated and patched on the next patch (or minor) release as soon as possible.
The project's security processes are described in [SECURITY.md](https://github.com/strimzi/.github/blob/main/SECURITY.md).

## Future Plans

### Functionality

There are several major features which we work on to further increase Strimzi capabilities.

Kafka itself is migrating from ZooKeeper for metadata management to its own Raft-based quorum, known as [KRaft][kip-500].
For Strimzi the challenge is ensure that users can access this new functionality and provide a migration path from existing ZooKeeper-based clusters.

To support more sophisticated use cases, we want to improve the manageability of Kafka cluster by integrating [Cruise Control][cc] more deeply within Strimzi.
We also intend to investigate enabling Kafka clusters to be spread over multiple Kubernetes clusters.

Finally, we see that there is a lot of value in leveraging the community's operational experience by providing common tooling, standard operating procedures and documentation.

> The Strimzi roadmap about new future features is available [here][https://github.com/orgs/strimzi/projects/1].

Reviewer 3: wrong format: should be `()` around URL, not `[]`

Strimzi can be installed using [Helm][helm] charts or using [OperatorHub.io][operatorhub] which is part of the [Operator Framework][operatorframework] project.

Strimzi joined CNCF Sandbox in September 2019

## Statement on alignment with the CNCF mission

> The Foundation's mission is to make cloud native computing ubiquitous

While Kafka is _capable_ of horizontal scalability, achieving such things as zero-downtime upgrades, and scaling clusters within Kubernetes without an operator is too high a barrier for many organizations.
This prevents ubiquitous use of Kafka anywhere with cloud native platforms such as Kubernetes.

Strimzi provides declarative APIs for Apache Kafka resources, including broker clusters, Kafka Connect clusters, connectors, and Kafka MirrorMaker instances (for inter-cluster replication).

Thanks to the above, Strimzi simplifies the way to deploy a Kafka cluster on Kubernetes as well as helps to automate the day-2 operations like upgrading brokers configuration, Kafka version and security related issues.

By automating the operational burdens of running Kafka, Strimzi brings this important technology into the hands of anyone using Kubernetes.

> 

Reviewer: Just a nit ... but no need for double empty line I guess?

# Strimzi Incubation Stage Proposal

## Introduction

Strimzi is a project focused on running [Apache Kafka][ak] on [Kubernetes][kube].
Apache Kafka is the _de facto_ standard platform for building real-time data pipelines.

> It provides support for high-throughput / low-latency messaging, as well as sophisticated development options that cover all the stages of a distributed data streaming pipeline, from ingestion, through storage to processing.

Reviewer 2: ```suggestion
It provides support for high-throughput / low-latency messaging, as well as sophisticated development options that cover all the stages of a distributed data streaming pipeline, from ingestion, through storage, and on to processing.
```

> But running it on Kubernetes can be complex and tedious.

Reviewer 2: ```suggestion
But running Kafka on Kubernetes can be complex and tedious.
```

> Strimzi makes it easy by using the operator pattern and declarative deployments.

Reviewer 2: ```suggestion
Strimzi makes it easy by using the Kubernetes operator pattern and declarative deployments.
```

> It provides operators not just for deploying Kafka clusters, but also for managing topics, users and connectors to make sure running Apache Kafka feels Kube-native.

Reviewer 2: ```suggestion
Strimzi provides operators not just for deploying Kafka clusters, but also for managing topics and users. Connectors help to make running Apache Kafka feel Kube-native.
```

Author: I don't think it works. `Connectors help to make running Apache Kafka feel Kube-native.` seems to mean that only connectors make Kafka feel kube native while it's all about topics, users and connectors. I think the current sentence makes more sense.

Strimzi also integrates with many other CNCF projects such as:

* Kubernetes
* [Prometheus][prom] for metrics and monitoring
* [Open Policy Agent][opa] for authorization
* [KEDA][keda] for autoscaling of Kafka clients
* [Jaeger][jaeger] and [OpenTelemetry][otel] for tracing

In addition, Strimzi is used by [Knative][knative] to provide its Kafka source.

> Strimzi can be installed using [Helm][helm] charts or using [OperatorHub.io][operatorhub] which is part of the [Operator Framework][operatorframework] project.

Reviewer 2: ```suggestion
Strimzi can be installed using [Helm][helm] charts or using [OperatorHub.io][operatorhub], which is part of the [Operator Framework][operatorframework] project.
```

Strimzi joined CNCF Sandbox in September 2019

## License

The Strimzi project is licensed under [Apache License 2.0](https://github.com/strimzi/strimzi-kafka-operator/blob/main/LICENSE).

## Statement on alignment with the CNCF mission

> The Foundation's mission is to make cloud native computing ubiquitous

While Kafka is _capable_ of horizontal scalability, achieving such things as zero-downtime upgrades, and scaling clusters within Kubernetes without an operator is too high a barrier for many organizations.
This prevents ubiquitous use of Kafka anywhere with cloud native platforms such as Kubernetes.

Strimzi provides declarative APIs for Apache Kafka resources, including broker clusters, Kafka Connect clusters, connectors, and Kafka MirrorMaker instances (for inter-cluster replication).

> Thanks to the above, Strimzi simplifies the way to deploy a Kafka cluster on Kubernetes as well as helps to automate the day-2 operations like upgrading brokers configuration, Kafka version and security related issues.

Reviewer 2: ```suggestion
Thanks to the declarative APIs, Strimzi simplifies the way to deploy a Kafka cluster on Kubernetes as well as helps to automate the day-2 operations like upgrading brokers configuration, Kafka version, and security related issues.
```

By automating the operational burdens of running Kafka, Strimzi brings this important technology into the hands of anyone using Kubernetes.

## Community

> In the Strimzi community users and developers can interact in several places:

Reviewer 2: ```suggestion
In the Strimzi community, users and developers can interact in several places:
```

* A public #strimzi channel, on the CNCF Slack workspace, with almost 1700 users.
* Mailing lists.
* Github issues and discussions.
* A public "community call" every other week. The full list with all recordings is [here](https://www.youtube.com/watch?v=wK4k3cAstbQ&list=PLpI4X8PMthYfONZopcRd4X_stq1C14Rtn).

> * We have participated in numerous KubeCon "Office Hours" events.

Reviewer 2: ```suggestion
* Forums and events. Strimzi has participated in numerous KubeCon "Office Hours" events.
```

Strimzi encourages contributions from the community, and has a good record of merging such contributions.

## Incubation Stage Requirements

<!-- NOTE: 
See [Incubation Stage](https://github.com/cncf/toc/blob/main/process/graduation_criteria.adoc#incubating-stage) requirements. -->

### Production usage

Strimzi is used in production by a number of organizations (see [ADOPTERS.md](https://github.com/strimzi/strimzi-kafka-operator/blob/main/ADOPTERS.md) for some public references).

> Some of the public adopters are also showed on the homepage of the official Strimzi [website](https://strimzi.io/).

Reviewer 2: ```suggestion
Some of the public adopters are presented on the homepage of the official Strimzi [website](https://strimzi.io/).
```

> In addition, the maintainers are aware of further non-public users of both Strimzi and commercially supported distributions of Strimzi. 

Reviewer 2: ```suggestion
In addition, Strimzi maintainers are aware of further non-public users of both Strimzi and commercially supported distributions of Strimzi. 
```

### Committers

> Strimzi has currently 7 maintainers from three companies (Red Hat, IBM and CN Group).

Reviewer 2: ```suggestion
Strimzi has currently 7 maintainers from three companies (Red Hat, IBM, and CN Group).
```

The full maintainer list can be found in our [GitHub repository](https://github.com/strimzi/governance/blob/master/MAINTAINERS).

> Additionally to the maintainers, 8 other contributors have _Write_ permissions to individual repositories and act as owners for the sub-projects within these repositories.

Reviewer 2: ```suggestion
In addition to the maintainers, 8 other contributors have _Write_ permissions to individual repositories and act as owners for the sub-projects within these repositories.
```

The full list can be found in our [GitHub repository](https://github.com/strimzi/governance/blob/main/COMPONENT-OWNERS) as well.

### Commits and merged contributions

At the time of writing:
* 2316 commits in the last 12 months, and more in the last 3 months than the previous 3 months
* 438 contributors in the last 12 months
* more than 3k stars and ~830 forks on github.

See our [devstats](https://all.devstats.cncf.io/d/54/project-health-table?orgId=1&var-repogroup_name=Strimzi) for more details.

### Versioning scheme

Strimzi uses a 1–2 month release schedule for the "core" operators component.
All Strimzi component releases use [semantic versioning](https://semver.org/).
The project's release version in described in [RELEASES.md](https://github.com/strimzi/governance/blob/main/RELEASES.md).

### Security processes

> Security is important to the Strimzi team and any vulnerabilities should be reported to Strimzi maintainers privately, to minimize attacks against current users of Strimzi before they are fixed.

Reviewer 2: ```suggestion
Security is taken seriously by the Strimzi team. Any vulnerabilities should be reported to Strimzi maintainers privately, to minimize potential attacks against current users of Strimzi before they are fixed.
```

Reported vulnerabilities will be investigated and patched on the next patch (or minor) release as soon as possible.
The project's security processes are described in [SECURITY.md](https://github.com/strimzi/.github/blob/main/SECURITY.md).

## Future Plans

### Functionality

> There are several major features which we work on to further increase Strimzi capabilities.

Reviewer 2: ```suggestion
There are several major features in development or under consideration that will further increase Strimzi capabilities.
```

Kafka itself is migrating from ZooKeeper for metadata management to its own Raft-based quorum, known as [KRaft][kip-500].
For Strimzi the challenge is ensure that users can access this new functionality and provide a migration path from existing ZooKeeper-based clusters.

> To support more sophisticated use cases, we want to improve the manageability of Kafka cluster by integrating [Cruise Control][cc] more deeply within Strimzi.

Reviewer 2: ```suggestion
To support more sophisticated use cases, we want to improve the manageability of Kafka clusters by integrating [Cruise Control][cc] more deeply within Strimzi.
```

Reviewer 2: Do we want to say how CC will help with this?

Author: I don't think we need to go into more technical details. The people reviewing the proposal won't need them. They are more interested in the overall "story", current adoption and future plans to see as the project is used in production, is mature enough and is also healthy from a community point of view.

We also intend to investigate enabling Kafka clusters to be spread over multiple Kubernetes clusters.

> Finally, we see that there is a lot of value in leveraging the community's operational experience by providing common tooling, standard operating procedures and documentation.

Reviewer 2: ```suggestion
Finally, we see that there is a lot of value in leveraging the community's operational experience by providing common tooling, standard operating procedures, and documentation.
```

> The Strimzi roadmap about new future features is available [here](https://github.com/orgs/strimzi/projects/1).

Reviewer 2: ```suggestion
The Strimzi roadmap of new future features is available [here](https://github.com/orgs/strimzi/projects/1).
```

### Usage and community

In the short to medium term we see two main drivers of adoption. 

Strimzi is already seeing significant production usage across many companies, but this is at odds with its current 0.x version number.
Announcing Strimzi 1.0 will help signal the software's maturity, especially to those who have been held back by an apparent lack of production-readiness. 

A second driver is the increased visibility of the project if Strimzi is accepted into CNCF incubation.
Taking Knative as an example, we would like to more actively promote Strimzi as a building block for other cloud-native technologies with data streaming requirements.

> We hope that increasing the number and diversity of organizations using Strimzi will have a corresponding effect on contributions to the project, and from this there is the prospect of growing the number of committers. 

Reviewer 2: ```suggestion
We hope that expanding the number and diversity of organizations using Strimzi will have a corresponding increase on the number of committers and contributions being made to the project.
```
